src/votekit/elections/election_types/ranking/stv.py

```diff
@@ -0,0 +1,406 @@
+from .abstract_ranking import RankingElection
+from ...transfers import fractional_transfer
+from ....pref_profile import PreferenceProfile
+from ...election_state import ElectionState
+from ....ballot import Ballot
+from ....utils import (
+    remove_cand,
+    first_place_votes,
+    ballots_by_first_cand,
+    tiebreak_set,
+    elect_cands_from_set_ranking,
+    score_dict_to_ranking,
+)
+from typing import Optional, Callable, Union
+from fractions import Fraction
+
+
+class STV(RankingElection):
+    """
+    STV elections. Currently implements simultaneous election. All ballots must have no
+    ties.
+
+    Args:
+        profile (PreferenceProfile):   PreferenceProfile to run election on.
+        m (int, optional): Number of seats to be elected. Defaults to 1.
+        transfer (Callable[[str, Union[Fraction, float], Union[tuple[Ballot], list[Ballot]], int], tuple[Ballot,...]], optional):
+        Transfer method. Defaults to fractional transfer.
+            Function signature is elected candidate, their number of first-place votes, the list of
+            ballots with them ranked first, and the threshold value. Returns the list of ballots
+            after transfer.
+        quota (str, optional): Formula to calculate quota. Accepts "droop" or "hare".
+            Defaults to "droop".
+        simultaneous (bool, optional): True if all candidates who cross threshold in a round are
+            elected simultaneously, False if only the candidate with highest first-place votes
+            who crosses the threshold is elected in a round. Defaults to True.
+        tiebreak (str, optional): Method to be used if a tiebreak is needed. Accepts
+            'borda' and 'random'. Defaults to None, in which case a ValueError is raised if
+            a tiebreak is needed.
+
+    """  # noqa
+
+    def __init__(
+        self,
+        profile: PreferenceProfile,
+        m: int = 1,
+        transfer: Callable[
+            [str, Union[Fraction, float], Union[tuple[Ballot], list[Ballot]], int],
+            tuple[Ballot, ...],
+        ] = fractional_transfer,
+        quota: str = "droop",
+        simultaneous: bool = True,
+        tiebreak: Optional[str] = None,
+    ):
+        self._stv_validate_profile(profile)
+
+        if m <= 0 or m > len(profile.candidates):
+            raise ValueError(
+                "m must be non-negative and less than or equal to the number of candidates."
+            )
+
+        self.m = m
+        self.transfer = transfer
+        self.quota = quota
+        self.threshold = 0  # for caching purposes
```

Comment:
```suggestion
        # Set to 0 initially so that first call to `get_threshold` returns the 
        # proper threshold.
        self.threshold = 0 
```